File: src/main/java/com/manning/javapersistence/ch08/java/repository/BidRepository.java
package com.manning.javapersistence.ch08.java.repository;

import com.manning.javapersistence.ch08.java.model.Bid;
import com.manning.javapersistence.ch08.java.model.Item;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Set;

public interface BidRepository extends JpaRepository<Bid, Long> {
    Set<Bid> findByItem(Item item);
}

File: src/main/kotlin/com/manning/javapersistence/ch08/kotlin/repository/BidRepository.kt
package com.manning.javapersistence.ch08.kotlin.repository

import com.manning.javapersistence.ch08.kotlin.model.Bid
import com.manning.javapersistence.ch08.kotlin.model.Item
import org.springframework.data.jpa.repository.JpaRepository

interface BidRepository : JpaRepository<Bid, Long> {
    fun findByItem(item: Item): Set<Bid>
}
